### PR TITLE
🐛 SingerSource: Fix incompatibilities and typing issues

### PR DESCRIPTION
**Pull request by cstruct**
## What
SingerSource has been incompatible with `split_config` as reported in #8710, the source has been full of typing errors, deprecated `AirbyteLogger` was used and method argument names has not been in line with parent classes. This PR fixes the incompatibility, removes usage of `AirbyteLogger`, method argument names are brought in line with parent classes, the typing errors are fixed and removes the exclude of singer when running type checking.

## How
**Incompatibility with `split_config`**:
`ConfigContainer` is made a subclass of `Dict[str, Any]` so `split_config` can work on it and we stop re-assembling the config after splitting it so the type is preserved.

**Type issues**:
`Connector` and `Source` are made generic to allow the singer source to specialise the types it uses for config, state and catalog. Both of them are also split into a base class where more methods are left as abstract and a class with the old name where these are implemented for the default types.

Usage of `AirbyteLogger` in method signatures are changed to `logging.Logger` since that is what is actually passed into these methods.

Remove exclude of singer from `mypy`.

**Bringing method argument names in line with parent classes**:
This isn't a typing fix since `mypy` allows renaming positional arguments but is done to keep the classes as similar as possible and avoid issues if they are ever called as keyword arguments.

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/connector.py`
2. `airbyte-cdk/python/airbyte_cdk/sources/source.py`
3. `airbyte-cdk/python/airbyte_cdk/sources/singer/source.py`
4. `airbyte-cdk/python/airbyte_cdk/sources/singer/__init__.py`
5. `airbyte-integrations/connector-templates/source-singer/source_{{snakeCase name}}_singer/source.py.hbs`
6. `airbyte-cdk/python/airbyte_cdk/entrypoint.py`
7. `airbyte-cdk/python/type_check_and_test.sh`

## 🚨 User Impact 🚨
If a singer source calls any method with keyword arguments where the name has changed it will break.

## Pre-merge Checklist
Unclear what checklist should be used for updates to the Python CDK.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by girarda — COMMENTED** [review]
love it! thank you for putting this together! I'll runs some safety checks and work on getting this released

File: airbyte-cdk/python/type_check_and_test.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # Static Type Checking
 echo "Running MyPy to static check and test files."
 # Exclude Singer for the time being.
```
Comment (girarda) [inline airbyte-cdk/python/type_check_and_test.sh:8]: nit: can you remove that deprecated comment?

Comment (cstruct) [reply airbyte-cdk/python/type_check_and_test.sh:8]: I'll remove it later this evening or tomorrow

**Comment by cstruct** [comment]
The one failing test looks to be unrelated to my changes.

**Review by girarda — APPROVED** [review]

**Pull request merged**